test/ConfigProviderTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DotTest\Rbac;
+
+use Dot\Rbac\Assertion\AssertionPluginManager;
+use Dot\Rbac\Authorization\AuthorizationService;
+use Dot\Rbac\ConfigProvider;
+use Dot\Rbac\Identity\AuthenticationIdentityProvider;
+use Dot\Rbac\Options\AuthorizationOptions;
+use Dot\Rbac\Role\Provider\RoleProviderPluginManager;
+use Dot\Rbac\Role\RoleService;
+use PHPUnit\Framework\TestCase;
+
+class ConfigProviderTest extends TestCase
+{
+    protected array $config;
+
+    protected function setup(): void
+    {
+        $this->config = (new ConfigProvider())();
+    }
+
+    public function testHasDependencies(): void
+    {
+        $this->assertArrayHasKey('dependencies', $this->config);
+    }
+
+    public function testHasDotAuthorization(): void
+    {
+        $this->assertArrayHasKey('dot_authorization', $this->config);
+    }
+
+    public function testDependenciesHasFactories(): void
```

Comment:
First, check if `dependencies` has `factories`:
`$this->assertArrayHasKey('factories', $this->config['dependencies']);`
only after that can `$factories` take the value of the `factories` key
`$factories = $this->config['dependencies']['factories'];`
Make sure you follow the same order with `testDependenciesHasAliases` method.